js/coinsbit.js

```diff
@@ -0,0 +1,1073 @@
+                        'account/order_history_list': 1,
+                    },
+                },
+            },
+            'fees': {
+                'trading': {
+                    'tierBased': false,
+                    'percentage': true,
+                    'maker': 0.2 / 100,
+                    'taker': 0.2 / 100,
+                },
+            },
+            'options': {
+                'createMarketBuyOrderRequiresPrice': true,
+                'fetchTradesMethod': 'publicGetHistory', // publicGetHistory | publicGetHistoryResult
+                'maxDiapasonsForTimeframes': {
+                    '1m': 43200,
+                    '30m': 86400,
+                    '1h': 86400 * 7,
+                    '1d': Number.MAX_SAFE_INTEGER,
+                },
+            },
+            'commonCurrencies': {
+                // 'xxx': 'xxxxxxx',
+            },
+            'exceptions': {
+                'exact': {
+                },
+                'broad': {
+                },
+            },
+        });
+    }
+
+    async fetchMarkets (params = {}) {
+        const response = await this.publicGetMarkets (params);
+        //     {
+        //         "success": true,
+        //         "message": "",
+        //         "result": [
+        //             {
+        //                 name: "BTC_USDT",
+        //                 moneyPrec: "8",
+        //                 stock: "BTC",
+        //                 money: "USDT",
+        //                 stockPrec: "8",
+        //                 feePrec: "8",
+        //                 minAmount: "0.001"
+        //             },
+        //             ...
+        //          ],
+        //         "code":"200",
+        //
+        const data = this.safeValue (response, 'result', []);
+        const result = [];
+        for (let i = 0; i < data.length; i++) {
+            const market = data[i];
+            const id = this.safeString (market, 'name');
+            const baseId = this.safeString (market, 'stock');
+            const quoteId = this.safeString (market, 'money');
+            const base = this.safeCurrencyCode (baseId);
+            const quote = this.safeCurrencyCode (quoteId);
+            const symbol = base + '/' + quote;
+            const priceScale = this.safeInteger (market, 'moneyPrec');
+            const quantityScale = this.safeInteger (market, 'stockPrec');
+            const pricePrecision = priceScale;
+            const quantityPrecision = quantityScale;
+            const minAmount = this.safeNumber (market, 'minAmount');
+            const type = 'spot';
+            result.push ({
+                'id': id,
+                'symbol': symbol,
+                'base': base,
+                'quote': quote,
+                'baseId': baseId,
+                'quoteId': quoteId,
+                'type': type,
+                'spot': true,
+                'derivative': undefined,
```

Comment:
we can remove the derivative property as it is deprecated,

we should also set contract to false, remove strike and optionType, and add `future: false`,

Reply:
> remove strike and optionType

Are you sure about those two?

Reply:
yep afaik we haven't got unified support for options anywhere, so i'd keep it out of the standard market structure for now

Reply:
> yep afaik we haven't got unified support for options anywhere, so i'd keep it out of the standard market structure for now

We are in the process of adding the support for options, including a number of recent fetchMarkets edits. So i would keep `strike` and `optionType`.